tests/tt_metal/tt_metal/dispatch/dispatch_util/test_ringbuffer_cache.cpp

```diff
@@ -0,0 +1,307 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+#include "tt_metal/impl/dispatch/ringbuffer_cache.hpp"
```

Comment:
I recommend we don't use raw pointers and we should use smart pointers instead so we don't have to deal with remembering to to delete

```suggestion
#include "tt_metal/impl/dispatch/ringbuffer_cache.hpp"
#include <memory>
```